src/Shared/NodePacketTranslator.cs

```diff
@@ -1414,6 +1484,103 @@ public static void DeserializeFields(object obj, List<SerializedField> serialize
             }
         }
 
+        public static class ProfilerResultTranslator
+        {
+            public static void Translate(INodePacketTranslator translator, ref ProfilerResult profilerResult)
+            {
+                if (translator.TranslateNullable(profilerResult))
+                {
+                    IDictionary<EvaluationLocation, ProfiledLocation> profiledLocations = new Dictionary<EvaluationLocation, ProfiledLocation>();
+                    if (translator.Mode == TranslationDirection.WriteToStream)
+                    {
+                        profiledLocations = profilerResult.ProfiledLocations.ToDictionary(kv => kv.Key, kv => kv.Value);
```

Comment:
This ToDictionary might take some time. How bad would it get if you made ProfilerResult.ProfiledLocations an ImmutableDictionary instead? That one implements IDictionary, and should not need the extra ToDictionary call.

Reply:
I usually prefer IReadOnlyDictionary since the immutability is clearer. Plus this is not in the hot path: it only takes place when the profiler is on, and after an evaluation is done. But if you feel strong about it, I don't mind changing it.

Reply:
let's leave it like this until it becomes a measurable problem :)